lib/project_types/script/layers/infrastructure/script_service.rb

```diff
@@ -146,6 +146,8 @@ def proxy_through_partners(query_name:, variables: nil, **options)
         end
 
         def raise_if_graphql_failed(response)
+          raise Errors::EmptyResponseError if response.nil?
```

Comment:
I wonder if it would make sense to catch this earlier in the stack, e.g in `ShopifyCli::PartnersAPI`?

Reply:
Yeah so the original exception is caught [in api.rb here](https://github.com/Shopify/shopify-app-cli/blob/d23176f1fc75cc1612f5f03a82d82f416a1cdabd/lib/shopify-cli/api.rb#L34) and we `ctx.puts` an error message and implicitly return nil for the response. Personally, I would prefer if we avoid catching any error in `api.rb` and instead leave it to the caller to catch any exception that they want handled, which is whats happening now for 400-499 response codes.  However, some of the code in extensions relies on the response being nil and I just didn't think it was worth the time or effort to change that right now 🤷‍♂️

Reply:
Yeah, it does seem that the current approach isn't ideal, so I suppose this is a reasonable way to handle it for now.